tests/Benchmarks/System.Text.JsonLab/JsonReaderPerf.cs

```diff
@@ -2,70 +2,277 @@
+                            }
+
+                            json.Value.CopyTo(destination);
+                            destination[json.Value.Length] = (byte)',';
+                            destination[json.Value.Length + 1] = (byte)' ';
+                            destination = destination.Slice(json.Value.Length + 2);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                return outputArray;
+            }
             else
-                return Encoding.UTF8.GetBytes(data);
+            {
+                Span<byte> destination = outputArray;
+                var json = new JsonReader(_dataUtf16, SymbolTable.InvariantUtf16);
+                while (json.Read())
+                {
+                    var tokenType = json.TokenType;
+                    switch (tokenType)
+                    {
+                        case JsonTokenType.PropertyName:
+                            json.Value.CopyTo(destination);
+                            destination[json.Value.Length] = (byte)',';
+                            destination[json.Value.Length + 2] = (byte)' ';
+                            destination = destination.Slice(json.Value.Length + 4);
+                            break;
+                        case JsonTokenType.Value:
+                            var valueType = json.ValueType;
+
+                            switch (valueType)
+                            {
+                                case JsonValueType.True:
+                                    destination[0] = (byte)'T';
+                                    destination[2] = (byte)'r';
+                                    destination[4] = (byte)'u';
+                                    destination[6] = (byte)'e';
+                                    destination = destination.Slice(8);
+                                    break;
+                                case JsonValueType.False:
+                                    destination[0] = (byte)'F';
+                                    destination[2] = (byte)'a';
+                                    destination[4] = (byte)'l';
+                                    destination[6] = (byte)'s';
+                                    destination[8] = (byte)'e';
+                                    destination = destination.Slice(10);
+                                    break;
+                            }
+
+                            json.Value.CopyTo(destination);
+                            destination[json.Value.Length] = (byte)',';
+                            destination[json.Value.Length + 2] = (byte)' ';
+                            destination = destination.Slice(json.Value.Length + 4);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                return outputArray;
+            }
```

Comment:
I would rather create a single method and make it parameterizable. When the code is being duplicated and we want to measure the difference we can get into a situation where the copies are not identical and the difference might be due some code mismatch. Not different perf

Reply:
How do you suggest we parameterize this? ArgumentSource? If so, how can we do the GlobalSetup correctly?